csharp/src/Ice/ConnectionI.cs

```diff
@@ -2906,6 +2945,7 @@ internal void completed(LocalException ex)
     private ConnectionInfo _info;
 
     private CloseCallback _closeCallback;
+    private readonly TaskCompletionSource _closed = new(); // can run synchronously
```

Comment:
See code where we call SetResult/SetException. The synchronous completion is consistent with the way we call the close callback, but maybe I am missing a potential pitfall?